smac_planner/src/a_star.cpp

```diff
@@ -130,13 +126,12 @@ bool AStarAlgorithm::areInputsValid()
   }
 
   // Check if ending point is valid
-  NodePtr node;
-  if (getToleranceHeuristic() < 0.001 && !isNodeValid(_goal->getIndex())) {
+  if (getToleranceHeuristic() < 0.001 && !_goal->isNodeValid(_traverse_unknown)) {
     throw std::runtime_error("Failed to compute path, goal is occupied with no tolerance.");
   }
 
   // Check if starting point is valid
-  if (!isNodeValid(getStart()->getIndex())) {
+  if (!getStart()->isNodeValid(_traverse_unknown)) {
```

Comment:
```suggestion
  if (!_start->isNodeValid(_traverse_unknown)) {
```

Reply:
Just because we set it here and we know this is synchronous, its a little faster

Reply:
Good catch, thanks.